tests/core/pyspec/eth2spec/test/eip7732/epoch_processing/test_process_builder_pending_payments.py

```diff
@@ -0,0 +1,301 @@
+from eth2spec.test.context import (
+    spec_state_test,
+    with_eip7732_and_later,
+)
+from eth2spec.test.helpers.epoch_processing import run_epoch_processing_with
+from eth2spec.test.helpers.state import next_epoch
+
+
+def create_builder_pending_payment(spec, builder_index, amount, weight=0, fee_recipient=None):
+    """Create a BuilderPendingPayment for testing."""
+    if fee_recipient is None:
+        fee_recipient = spec.ExecutionAddress()
+
+    return spec.BuilderPendingPayment(
+        weight=weight,
+        withdrawal=spec.BuilderPendingWithdrawal(
+            fee_recipient=fee_recipient,
+            amount=amount,
+            builder_index=builder_index,
+            withdrawable_epoch=spec.FAR_FUTURE_EPOCH,
+        ),
+    )
+
+
+@with_eip7732_and_later
+@spec_state_test
+def test_process_builder_pending_payments_empty_queue(spec, state):
+    """Test processing with no pending payments."""
+    # Advance past genesis epochs
+    next_epoch(spec, state)
+    next_epoch(spec, state)
+
+    # Populate the first SLOTS_PER_EPOCH with payments to ensure they're not empty
+    for i in range(spec.SLOTS_PER_EPOCH):
+        payment = create_builder_pending_payment(spec, i, spec.MIN_ACTIVATION_BALANCE, 1)
+        state.builder_pending_payments[i] = payment
+
+    pre_builder_pending_withdrawals = len(state.builder_pending_withdrawals)
+
+    yield from run_epoch_processing_with(spec, state, "process_builder_pending_payments")
+
+    # No withdrawals should be added
+    assert len(state.builder_pending_withdrawals) == pre_builder_pending_withdrawals
+
+    # Queue should be rotated - first SLOTS_PER_EPOCH should be empty
+    for i in range(spec.SLOTS_PER_EPOCH):
+        payment = state.builder_pending_payments[i]
+        assert payment.weight == 0
+        assert payment.withdrawal.amount == 0
+        assert payment.withdrawal.builder_index == 0
+
+
+@with_eip7732_and_later
+@spec_state_test
+def test_process_builder_pending_payments_below_quorum(spec, state):
+    """Test payment below quorum threshold - should not be processed."""
+    # Advance past genesis epochs
+    next_epoch(spec, state)
+    next_epoch(spec, state)
+
+    builder_index = 0
+    amount = spec.MIN_ACTIVATION_BALANCE
+    quorum = spec.get_builder_payment_quorum_threshold(state)
+    weight = quorum // 2  # Below threshold
```

Comment:
I'd prefer to do the following for better bounds checking.
```suggestion
    weight = quorum - 1  # Below threshold
```